lib/real-device.js

```diff
@@ -71,82 +69,50 @@ export class RealDevice {
 
   /**
    *
-   * @param {string} app
-   * @param {number} timeout
-   * @param {'ios-deploy'|'serial'|'parallel'|null} strategy
-   * @privateRemarks This really needs type guards built out
+   * @param {string} appPath
+   * @param {string} bundleId
+   * @param {InstallOptions} [opts={}]
    */
-  async install(app, timeout, strategy = null) {
-    if (
-      strategy &&
-      !_.values(APP_INSTALL_STRATEGY).includes(/** @type {any} */ (_.toLower(strategy)))
-    ) {
-      throw new Error(
-        `App installation strategy '${strategy}' is unknown. ` +
-          `Only the following strategies are supported: ${_.values(APP_INSTALL_STRATEGY)}`,
-      );
-    }
-    this.log.debug(
-      `Using '${strategy ?? APP_INSTALL_STRATEGY.SERIAL}' app deployment strategy. ` +
-        `You could change it by providing another value to the 'appInstallStrategy' capability`,
-    );
-
-    const installWithIosDeploy = async () => {
-      try {
-        await fs.which(IOS_DEPLOY);
-      } catch (err) {
-        throw new Error(`'${IOS_DEPLOY}' utility has not been found in PATH. Is it installed?`);
-      }
-      try {
-        await exec(IOS_DEPLOY, ['--id', this.udid, '--bundle', app], {timeout});
-      } catch (err) {
-        throw new Error(err.stderr || err.stdout || err.message);
-      }
-    };
-
+  async install(appPath, bundleId, opts = {}) {
+    const {
+      timeoutMs = IO_TIMEOUT_MS,
+    } = opts;
     const timer = new timing.Timer().start();
-    if (_.toLower(/** @type {'ios-deploy'} */ (strategy)) === APP_INSTALL_STRATEGY.IOS_DEPLOY) {
-      await installWithIosDeploy();
-    } else {
-      const afcService = await services.startAfcService(this.udid);
-      const enableParallelPush = _.toLower(/** @type {'parallel'} */ (strategy)) === APP_INSTALL_STRATEGY.PARALLEL;
-      try {
-        const bundleId = await extractBundleId(app);
-        const bundlePathOnPhone = path.join(INSTALLATION_STAGING_DIR, bundleId);
-        await pushFolder(afcService, app, bundlePathOnPhone, {
-          enableParallelPush,
-          timeoutMs: timeout,
+    const afcService = await services.startAfcService(this.udid);
+    try {
+      let bundlePathOnPhone;
+      if ((await fs.stat(appPath)).isFile()) {
+        // https://github.com/doronz88/pymobiledevice3/blob/6ff5001f5776e03b610363254e82d7fbcad4ef5f/pymobiledevice3/services/installation_proxy.py#L75
+        bundlePathOnPhone = `/${path.basename(appPath)}`;
+        await pushFile(afcService, appPath, bundlePathOnPhone, {
```

Comment:
not this pr change itself, but can we add log like `Successfully pushed 303 folders and 742 files within 12807ms` in pushFolder, how long did the push file take, as debug log info? Such log helped in the past

Reply:
added this now